lib/modules/manager/pub/utils.ts

```diff
@@ -0,0 +1,16 @@
+import { Lazy } from '../../../util/lazy';
+import { PubspecLockSchema } from './schema';
+
+export function lazyParsePubspeckLock(
+  fileContent: string
+): Lazy<PubspecLockSchema | undefined> {
+  return new Lazy(() => parsePubspecLock(fileContent));
+}
+
+function parsePubspecLock(fileContent: string): PubspecLockSchema | undefined {
+  const res = PubspecLockSchema.safeParse(fileContent);
```

Comment:
I updated the PR to use zod as suggested, but I'm having trouble with it here. Basically, I'm not sure how I should convert the file content string into an object for zod to validate.

Reply:
use yaml-js to parse yaml to object